src/mango_idx.erl

```diff
@@ -134,6 +135,50 @@ remove(DDoc, Idx) ->
     {ok, NewDDoc#doc{body = Body}}.
 
 
+%% If one of the design doc names happens to be a non-query design doc,
+%% mango_util:load_ddoc throw an error before bulk delete.
+bulk_delete(Db, DDocIds, DelOpts0) ->
+    DelOpts = mango_crud:maybe_add_user_ctx(Db, DelOpts0),
+    DeleteDocs = lists:map(fun(DDocId0) ->
+        DDocId = case DDocId0 of
+            <<"_design/", _/binary>> -> DDocId0;
+            _ -> <<"_design/", DDocId0/binary>>
+        end,
+        {ok, DDoc} = mango_util:load_ddoc(Db, DDocId),
+        DDoc#doc{deleted = true, body = {[]}}
+    end, DDocIds),
+    case fabric:update_docs(Db, DeleteDocs, DelOpts) of
+        {ok, Results} ->
+            bulk_delete_results(lists:zip(DeleteDocs, Results));
+        {accepted, Results} ->
+            bulk_delete_results(lists:zip(DeleteDocs, Results));
+        {aborted, Errors} ->
+            bulk_delete_results(lists:zip(DeleteDocs, Errors))
+    end.
+
+
+%% Iterate through each ddocid and the associated result. Put the ddocid
+%% in the proper Success or Fail list to return to user.
+bulk_delete_results(DeleteResults) ->
+    lists:foldl(fun({#doc{id=DDocId}, Result}, {Success, Fail}) ->
+        Id = {<<"id">>, DDocId},
+        case Result of
+            % This means we deleted a doc that didn't exist.
```

Comment:
This should be impossible since you already loaded the ddoc up above. Also it would be a not good thing to actually make a write if the ddoc didn't exist.